src/azure-cli/azure/cli/command_modules/sql/custom.py

```diff
@@ -319,12 +341,18 @@ def _db_elastic_pool_update_sku(
     if capacity:
         instance.sku.capacity = capacity
 
+    # Wipe out sku name if serverless vs provisioned db offerings changed,
+    # only if sku name has not be wiped by earlier logic, and new compute model has been requested.
+    if instance.sku.name is not None and compute_model is not None:
```

Comment:
It's more pythonic to implicitly convert null check to boolean, i.e. `if instance.sku.name and compute_model:`